BaseSpace.SDK/Infrastructure/FileUpload.cs

```diff
@@ -196,18 +196,33 @@ public static int NumFileUploadParts(FileInfo file)
         }
 
         private class BSWebClient : WebClient
-		{
-			//TODO: Doesnt seem right? Need refactor?
-			// const int CONNECTION_LIMIT = 16;
-			protected override WebRequest GetWebRequest(Uri address)
-			{
-				var request = base.GetWebRequest(address) as HttpWebRequest;
-
-				if (request != null && request.ServicePoint.ConnectionLimit < 20)
-					request.ServicePoint.ConnectionLimit = 10000;  //Note: Is this changing global value?
-
-				return request;
-			}
-		}
+        {
+            //TODO: Doesnt seem right? Need refactoring?
+            // const int CONNECTION_LIMIT = 16;
```

Comment:
The actual changes are just on line 220 and 221 - increasing the default timeout.  Note that the default chunk size is 25mb, making this even a bigger issue on slow connections.

The 500 errors mentioned in the comments should probably also be looked at.  Can you relay this potential issue to the team responsible for this?